### PR TITLE
Add SWA warning if not running every epoch

### LINKED ISSUE #6963: Turning SWA on makes scheduler lr change to epoch, instead of batch [with colab ex]
## 🐛 Bug

Below is my optimizer/scheduler code. If my trainer has `stochastic_weight_avg=True`, then my learning rate is shown below, in green, and I get the warning:

_/home/felipe/anaconda3/envs/ML_38_new/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py:68: UserWarning: Swapping lr_scheduler <torch.optim.lr_scheduler.OneCycleLR object at 0x7fa445c76ee0> for <torch.optim.swa_utils.SWALR object at 0x7fa445e2d190>
  warnings.warn(*args, **kwargs)_

If `stochastic_weight_avg=False`, then I get the appropriate learning rate scheduler (pink).

image

What seems to me is that when `stochastic_weight_avg=True`, there is some conflict related to updating per batch or per epoch

```
def configure_optimizers(self):
        params = list(self.named_parameters())
        def is_backbone(n): return 'encoder' in n
        
        grouped_parameters = [
            {'params': [p for n, p in params if is_backbone(n)], 'lr': cfg.max_lr/cfg.encoder_lr_frac},
            {'params': [p for n, p in params if not is_backbone(n)], 'lr': cfg.max_lr},
        ]

        optimizer = MADGRAD(grouped_parameters, lr=cfg.max_lr, weight_decay=cfg.wd)
        
        if cfg.scheduler_type == 'onecycle':
            scheduler_fn = torch.optim.lr_scheduler.OneCycleLR(optimizer,
                                                                max_lr=[cfg.max_lr/cfg.encoder_lr_frac, cfg.max_lr],
                                                                epochs=self.epochs,
                                                                steps_per_epoch = len(self.loaders_dict['train']))
            
        scheduler = {'scheduler': scheduler_fn,
                     'name': cfg.scheduler_type,
                     'frequency': 1,
                     'interval': 'step',
                     "monitor": 'train_loss'}#len(self.loaders_dict['train']))
            
        return [optimizer], [scheduler]
```

### To Reproduce
https://gist.github.com/fmellomascarenhas/7e53efbacaafd8769088d58574e73cd5

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Adds a warning to docs and when running
Fixes tests issue where the lr_scheduler was not being set

Fixes #6963

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by kaushikb11 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/callbacks/stochastic_weight_avg.py [hunk 1]
```diff
@@ -187,16 +189,18 @@ def on_train_epoch_start(self, trainer: 'pl.Trainer', pl_module: 'pl.LightningMo
                 anneal_strategy=self._annealing_strategy,
                 last_epoch=trainer.max_epochs if self._annealing_strategy == "cos" else -1
             )
-            _scheduler_config = _get_default_scheduler_config()
-            assert _scheduler_config["interval"] == "epoch" and _scheduler_config["frequency"] == 1
-            _scheduler_config["scheduler"] = self._swa_scheduler
+            default_scheduler_cfg = _get_default_scheduler_config()
+            assert default_scheduler_cfg["interval"] == "epoch" and default_scheduler_cfg["frequency"] == 1
```
Comment (ethanwharris) [inline pytorch_lightning/callbacks/stochastic_weight_avg.py:193]: n00b question: why is this assert statement here? `_get_default_scheduler_config` just returns a `dict` right? Also, if it can sometimes not return a dict, why give error sometimes and warning others?

Comment (carmocca) [reply pytorch_lightning/callbacks/stochastic_weight_avg.py:193]: it will return a dict. the assert is there as a sanity check in case somebody modifies it. But it could be removed

Comment (ethanwharris) [reply pytorch_lightning/callbacks/stochastic_weight_avg.py:193]: Makes sense :smiley:
